subprojects/launcher/src/test/groovy/org/gradle/tooling/internal/provider/serialization/ClasspathInfererTest.groovy

```diff
@@ -57,6 +57,18 @@ class ClasspathInfererTest extends AbstractClassGraphSpec {
         action.execute(null)
     }
 
+    def "determines action and tooling API classpath when loaded via custom faulty ClassLoader implementation"() {
```

Comment:
"faulty" is not the right term here. We should just say explicitly what it is: "With spaces in the classpath"